backends/p4test/CMakeLists.txt

```diff
@@ -106,6 +106,9 @@ set (P4_XFAIL_TESTS
   testdata/p4_16_samples/pna-example-tcp-connection-tracking-err-1.p4
   testdata/p4_16_samples/pna-example-tcp-connection-tracking-err.p4
   testdata/p4_16_samples/pna-example-pass-2.p4
+  testdata/p4_16_samples/invalid-hdr-warnings6.p4
+  testdata/p4_16_samples/invalid-hdr-warnings7.p4
```

Comment:
Handling validity flags for header union instances introduces if condition. These case have started failing because the header union instance is present within action and conditional statements are not supported within action for p4test backend. Perhaps, we should enable the support of conditional statements within action for p4test as well.

Reply:
the frontend has no problem with that, we need code generation in the back-end.
You could try the ebpf backend, I think that one supports already conditions in actions.
For bmv2 we never got around to implementing that, although there is support.

Reply:
DPDK also supports conditions in action and some of these tests are ported to PNA and are added to the DPDK regressions. 
We could try enabling this pass for other backends . However, I would prefer to limit this PR to dpdk and p4test backends.